MetaMask/metamask-mobile
# fix: Onboarding failing biometrics locks screen for user instead of disabling biometrics and continuing with the onboarding

## **Description**

During the course of onboarding, if the user enables Face ID to log in to the app, but the Face ID is unsuccessful in reading the users face, or the user cancels the Face ID check, the user is then locked out of the app and kicked out of the onboarding flow. When they sign back in they are taken to the wallet screen.

I have added a fix so that if the Face ID fails or the user cancels the Face ID check, Face ID will be disabled and the user will have to log in with their password. The onboarding process will also continue without locking the user out of the app.

The fix isn't 'ideal' as we use a `catch` block to catch an error and then perform some state setting, when ideally a `catch` should be used exclusively for error handling. However, when Biometrics fail, React Native throws an error that ends up in the `catch`. I also would prefer an `enum` to be used but we are in a JS file instead of a TS file.

To test the issue repeat the following:

1. Install the app on device.
2. Start the onboarding and enter a new password.
3. Enable Face ID (iOS) and cover the camera so the Face ID fails.
4. Then hit cancel, or repeat the process and hit cancel when the second alert appears.
5. You should then be pushed through to the next onboarding screen and you should not be sent back to the lock screen and be asked to log in again.

## **Related issues**

Fixes: https://github.com/orgs/MetaMask/projects/60/views/16?pane=issue&itemId=84422135&issue=MetaMask%7Cmetamask-mobile%7C11964

## **Manual testing steps**

1. Install the app on device.
2. Start the onboarding and enter a new password.
3. Enable Face ID (iOS) and cover the camera so the Face ID fails.
4. Then hit cancel, or repeat the process and hit cancel when the second alert appears.
5. You should then be pushed through to the next onboarding screen and you should not be sent back to the lock screen and be asked to log in again.

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/39dc4b1f-3099-433f-ac7b-4aff9e6ca742

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
